.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,115 @@
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+          submodules: true
+
+      - name: Install dependencies
+        run: |
+          apt-get update && apt-get install -y lcov libusb-1.0-0 libudev-dev cmake g++ python3 python3-pip
+          pip3 install ledgerblue --break-system-packages
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
```

actionlint (1.7.7)

the runner of "actions-rs/toolchain@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)